src/java.base/share/classes/sun/security/rsa/SunRsaSignEntries.java

```diff
@@ -56,53 +61,56 @@ public SunRsaSignEntries(Provider p) {
+                    "sun.security.rsa.RSAKeyFactory$Legacy",
+                    rsaAliases, null);
+            add(p, "KeyPairGenerator", "RSA",
+                    "sun.security.rsa.RSAKeyPairGenerator$Legacy",
+                    rsaAliases, null);
+            add(p, "Signature", "MD2withRSA",
+                    "sun.security.rsa.RSASignature$MD2withRSA",
+                    createAliasesWithOid(rsaOid + ".2"), attrs);
+            add(p, "Signature", "MD5withRSA",
+                    "sun.security.rsa.RSASignature$MD5withRSA",
+                    createAliasesWithOid(rsaOid + ".4"), attrs);
+            add(p, "Signature", "SHA1withRSA",
+                    "sun.security.rsa.RSASignature$SHA1withRSA",
+                    createAliasesWithOid(rsaOid + ".5", sha1withRSAOid2), attrs);
+            add(p, "Signature", "SHA224withRSA",
+                    "sun.security.rsa.RSASignature$SHA224withRSA",
+                    createAliasesWithOid(rsaOid + ".14"), attrs);
+            add(p, "Signature", "SHA256withRSA",
+                    "sun.security.rsa.RSASignature$SHA256withRSA",
+                    createAliasesWithOid(rsaOid + ".11"), attrs);
+            add(p, "Signature", "SHA384withRSA",
+                    "sun.security.rsa.RSASignature$SHA384withRSA",
+                    createAliasesWithOid(rsaOid + ".12"), attrs);
+            add(p, "Signature", "SHA512withRSA",
+                    "sun.security.rsa.RSASignature$SHA512withRSA",
+                    createAliasesWithOid(rsaOid + ".13"), attrs);
+            add(p, "Signature", "SHA512/224withRSA",
+                    "sun.security.rsa.RSASignature$SHA512_224withRSA",
+                    createAliasesWithOid(rsaOid + ".15"), attrs);
+            add(p, "Signature", "SHA512/256withRSA",
+                    "sun.security.rsa.RSASignature$SHA512_256withRSA",
+                    createAliasesWithOid(rsaOid + ".16"), attrs);
+
+            add(p, "KeyFactory", "RSASSA-PSS",
+                    "sun.security.rsa.RSAKeyFactory$PSS",
+                    rsapssAliases, null);
+            add(p, "KeyPairGenerator", "RSASSA-PSS",
+                    "sun.security.rsa.RSAKeyPairGenerator$PSS",
+                    rsapssAliases, null);
+            add(p, "Signature", "RSASSA-PSS",
+                    "sun.security.rsa.RSAPSSSignature",
+                    rsapssAliases, attrs);
+        }
 
-        add(p, "KeyFactory", "RSASSA-PSS",
-                "sun.security.rsa.RSAKeyFactory$PSS",
-                rsapssAliases, null);
-        add(p, "KeyPairGenerator", "RSASSA-PSS",
-                "sun.security.rsa.RSAKeyPairGenerator$PSS",
-                rsapssAliases, null);
-        add(p, "Signature", "RSASSA-PSS",
-                "sun.security.rsa.RSAPSSSignature",
-                rsapssAliases, attrs);
         add(p, "AlgorithmParameters", "RSASSA-PSS",
                 "sun.security.rsa.PSSParameters",
                 rsapssAliases, null);
```

Comment:
Luckily, `attrs` isn't used in this call, so we don't have the problem described for `SunEntries.java`. We could even move the `attrs` creation inside the `if (!systemFipsEnabled)` body with no issues, but I don't think we need to do that.

Reply:
Yes, I agree with your assessment here.

Reply:
Actually, it looks to me that the following entry in SunRsaSignEntries.java is affected in the sense that it does not have an attribute set in FIPS mode:

`addA(p, "KeyFactory", "RSASSA-PSS", "sun.security.rsa.RSAKeyFactory$PSS", attrs);`

I'll unresolve this conversation.

Reply:
I think you are referring to the [17u version](https://github.com/rh-openjdk/jdk/blob/6e74f283739af0d867df01d20f82865f559a45ea/src/java.base/share/classes/sun/security/rsa/SunRsaSignEntries.java#L102-L103). Here, `attrs` is passed as `null`:
https://github.com/rh-openjdk/jdk11u/blob/4a9d3e11048967658a2c9355e6056bb5796c2f34/src/java.base/share/classes/sun/security/rsa/SunRsaSignEntries.java#L97-L99

Reply:
As commented above, I'll resolve this conversation and backport RH2134669 to 11u separately.